Translate/open translation toggle in publishing tools; done for articles AND pages

jacqui opened the pull request:
Closes #320 
Closes #326 

**Update (Thursday): this  functionality now may be tested with 👉 version 94 👈  in the script editor; includes translations for pages**

**Update (Friday): now displays full locale name, prevents dupe buttons, and opens new translated doc in a new tab in one step**

Based on the locales available for a given organisation and what google documents are associated with an article, this PR updates the publishing tools so, for the Test Diaryo site that publishes in english and filipino:

* a document for an article in english only will display a "Translate to Filipino" button
* upon clicking that "translate" button, a new document will be created and the button updated to "open translation"
* clicking to open will open the new doc in a new browser tab
* subsequent loads of the original english document sidebar will display the "Open in fil" button that opens the translated doc in another tab

So, for each additional locale that a site publishes in:

* if a google doc exists for this article, display an "open" button for the translation
* if no google doc exists, display a "translate" button to create a new translation

This gets a little confusing to test with the code only in the script editor testing mode: opening the sidebar in the newly created translation doc will not show the right buttons as this PR hasn't been published for the add-on yet!

Making all of this work was a bit easier this time around for me, but it's still kinda tricky, so I appreciate the help testing it out.

TylerFisher commented:
Cool, this all works! I tested moving between translations that are already published, and I created a new set of articles that are connected via translation. I have a one very small edit about how this works, then I think this is good to go:

When I originally published my English translation, a button to translate didn't appear until I closed and reopened the sidebar. Is it possible for that button to appear right after the first publish? I can imagine people having both translations in hand and wanting to publish them immediately. 

And a couple of thoughts, not sure if we need to change anything though:
- I'm curious whether people will understand the locale codes we use everywhere, or if we should keep labels around for those. English instead of `en`, Spanish instead of `es`, Tagalog instead of `tl`, etc.
- I think I remember that this isn't possible, but is it possible to open the translation in another tab when you first click translate, instead of the double click thing?

jacqui commented:
> Cool, this all works! I tested moving between translations that are already published, and I created a new set of articles that are connected via translation. I have a one very small edit about how this works, then I think this is good to go:
> 
> When I originally published my English translation, a button to translate didn't appear until I closed and reopened the sidebar. Is it possible for that button to appear right after the first publish? I can imagine people having both translations in hand and wanting to publish them immediately.
> 
> And a couple of thoughts, not sure if we need to change anything though:
> 
>     * I'm curious whether people will understand the locale codes we use everywhere, or if we should keep labels around for those. English instead of `en`, Spanish instead of `es`, Tagalog instead of `tl`, etc.
> 
>     * I think I remember that this isn't possible, but is it possible to open the translation in another tab when you first click translate, instead of the double click thing?


Good points all around. Thanks for testing this out so thoroughly.

I've updated this PR as follows:

* the translate button now appears once you preview or publish the content
* the full locale name is now used in the button - I meant to change that before and forgot to update all the queries to include both the code and locale, should be covered now
* I also ran into multiple dupe buttons showing up while taking care of the above, so I added code to prevent that from happening (aka using unique locale-specific ids on the buttons)
 
I'll see about getting the translate functionality opening a new window in one step now. I did have trouble with getting that to work in the past though, you're right.
